lib/house.rb

```diff
@@ -0,0 +1,50 @@
+class House
+  attr_reader :address, :rooms
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @market_average = 500000
+  end
+
+  def price
+    (@price.delete("$")).to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    return true if price > @market_average
+      false
+  end
+
+  def rooms_from_category(category)
+    rooms.find_all do |room|
+      category == room.category
+    end
+  end
+
+  def area
+    room_areas = rooms.map { |room| room.area }
+    room_areas.sum
+  end
+
+  def details
+    details = Hash.new
+    details["price"] = price
+    details["address"] = address
+    details
```

Comment:
very nicely done

Reply:
I was saved by some of the Ruby Exercises that forced me to work with hashes.  I don't feel comfortable with them yet, but I had enough exposure that I didn't totally freak out!

Reply:
yeah... ideally, we would have taught hashes before this independent challenge, but... ❄️